### DISCUSSION TITLE
language specialized Copilots

### DISCUSSION THREAD

**Discussion by baboon25** [community/community#116422] [Copilot Conversations]
### Select Topic Area

General

### Body

I think it would be great to have language specialized copilots that know the ins and outs of a programming language. These should be trained on an up-to-date input stream of official language sources and the most popular libraries and related repositories, and therefore always have access to the latest language developments.

**Why?**
I start to notice that the knowledge of agents deprecates quite fast. For example, I'm currently using the copilot chat to help me with Rust problems and the agent regularly tells me, that the current Rust version is 1.57 while in reality it is, of this writing, 1.77. That's more than two years in between.
Similar things happen with its knowledge of certain repositories too.
With languages, that are progressing and changing at these rapid speeds, the current Copilot just doesn't hold up.

**Comment by Tgentil** [comment]
Your idea for language-specialized Copilots trained on current sources is insightful. Currently, GitHub Copilot is based on OpenAI's model trained on a dataset that includes code up until early 2021, so it doesn't automatically update with the latest language versions or libraries. However, OpenAI recently introduced a feature in ChatGPT that allows for training customized models. It might offer a solution for your need. Alternatively, you might find community-created models updated with recent language developments. Keeping an eye on official announcements from GitHub or OpenAI for future updates to Copilot could also be beneficial.
